matter_server/server/device_controller.py

```diff
@@ -403,8 +405,16 @@ async def open_commissioning_window(
         if self.chip_controller is None:
             raise RuntimeError("Device Controller not initialized.")
 
+        if (node := self._nodes.get(node_id)) is None or not node.available:
+            raise NodeNotReady(f"Node {node_id} is not (yet) available.")
+
+        if node_id in self._known_commissioning_params:
+            # node has already been put into commissioning mode,
+            # return previous parameters
+            return self._known_commissioning_params[node_id]
+
         if discriminator is None:
-            discriminator = 3840  # TODO generate random one
+            discriminator = randint(1000, 4096)  # noqa: S311
```

Comment:
Any reason to start from 1000? Also the constraint according to the OpenCommissioningWindow is to be between 0 and 4095, so:

```suggestion
            discriminator = randint(0, 4095)  # noqa: S311
```